src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -38,7 +38,14 @@ public abstract class LuceneDocumentConverterBase : IDisposable
 
         protected LuceneDocumentConverterBase(ICollection<IndexField> fields, bool reduceOutput = false)
         {
-            _fields = fields.ToDictionary(x => x.Name, x => x, StringComparer.OrdinalIgnoreCase);
+            _fields = fields.ToDictionary(x => x.Name, x =>
```

Comment:
Yes, that is different. In 3.5 we stored the map-reduce result as follow:
- each field of the output separately according to `IndexField` options
- entire output was stored in additional field under `__reduced_val` key in JSON format

In 4.0 we store each field separately and mark it as stored in order to be able to retrieve the result on query. This way we don't have the additional field and avoid parsing.